declfilter/declfilter.go

```diff
@@ -33,69 +105,49 @@ type DeclFilter interface {
 	HasTrusted() bool
 }
 
-type declFilter struct {
-	isTrivial bool // trivial filter translates everything and has no axioms.
-
-	toImport     map[string]bool
-	toTrust      map[string]bool
-	toTranslate  map[string]bool
-	toAxiomatize map[string]bool
-}
-
 func (df *declFilter) GetAction(name string) Action {
 	switch {
-	case df.isTrivial, df.toTranslate[name]:
+	case df.toTranslate.contains(name):
 		return Translate
-	case df.toAxiomatize[name]:
+	case df.toAxiomatize.contains(name):
 		return Axiomatize
-	case df.toTrust[name]:
+	case df.trusted.contains(name):
 		return Trust
 	default:
 		return Skip
 	}
 }
 
 func (df *declFilter) ShouldImport(i string) bool {
-	return df.isTrivial || df.toImport[i]
+	return df.imports.contains(i)
 }
 
 func (df *declFilter) HasTrusted() bool {
-	return len(df.toTrust) > 0
+	return len(df.trusted) > 0
 }
 
 func Load(raw []byte) DeclFilter {
```

Comment:
Can you factor out the FilterConfig -> DeclFilter conversion and add a couple tests for it? There's a lot we could test but I think one or two tests should be in this PR and we can add more later if someone revisits this code. You could also directly test this function but I personally think splitting it is nice for testing if nothing else.